persist/sqlite/consensus_refactored_test.go

```diff
@@ -2558,6 +2558,70 @@ func TestHostScan(t *testing.T) {
 	assertHost(hosts[0].PublicKey, host1)
 }
 
+func TestEventPayoutContract(t *testing.T) {
+	// test to catch bug where slice returned by explorer.AppliedEvents did not
+	// include miner payout events if there was any contract action in the
+	// block besides resolutions because it mistakenly returned early
+	pk1 := types.GeneratePrivateKey()
+	uc1 := types.StandardUnlockConditions(pk1.PublicKey())
+	addr1 := uc1.UnlockHash()
+
+	pk2 := types.GeneratePrivateKey()
+	uc2 := types.StandardUnlockConditions(pk2.PublicKey())
+	addr2 := uc2.UnlockHash()
+
+	n := newTestChain(t, false, func(network *consensus.Network, genesisBlock types.Block) {
+		genesisBlock.Transactions[0].SiacoinOutputs[0].Address = addr1
+	})
+	genesisTxn := n.genesis().Transactions[0]
+
+	fc := prepareContract(addr1, n.tipState().Index.Height+1)
+	// create file contract
+	txn1 := types.Transaction{
```

Comment:
could you also add a v2 txn to make sure the second occurrence gets hit as well?